test/new-e2e/tests/agent-metrics-logs/log-agent/utils/file_tailing_utils.go

```diff
@@ -168,6 +168,32 @@ func CheckLogsExpected(t *testing.T, fakeIntake *components.FakeIntake, service,
 	}, 2*time.Minute, 10*time.Second)
 }
 
+// CheckNoDuplicateTags verifies that there is no duplicate tags
+func CheckNoDuplicateTags(t *testing.T, fakeIntake *components.FakeIntake, service, content string) {
```

Comment:
Would it make sense to instead update the above `CheckLogsExpected` to check for duplicate tags? 
That way we get dupe tag coverage for all test cases instead of just one. I don't think there is a good reason why we couldn't perform this check on all test cases.